ENH: crosstabs analysis added to autolysis

skathirmani opened the pull request.

skathirmani commented:
@pratapvardhan: can we add the module `orderedattrdict` to travis dependency?. The test is failing because of that

pratapvardhan commented:
Can you add orderedattrdict to requirements.txt, travis will also be taken care of. And, please squash the commits, if you can.

pratapvardhan commented:
@skathirmani could you also confirm if the docstring for crosstab(s) are upto date?

skathirmani commented:
@pratapvardhan I have squashed the commits and added `orderedattrdict` to `requirements.txt`. I have cross checked the `crosstabs` doc strings at high level. Anyway I have to add detailed description about the output, which I will be doing later.. Requesting you to review and merge the commit

pratapvardhan commented:
Failure is due to lack of `orderedattrdict` on conda channel. 

cc @sanand0

pratapvardhan commented on autolysis/__init__.py:
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
+    Any value above 0.5 is a strong indication of a relationship. This says
+    that ``language`` and ``city`` are dependent.
+    '''
+    observed = pd.crosstab(index, column, values, aggfunc='sum').replace('', np.nan).fillna(0)
```
line too long, could we multi-line it?

pratapvardhan commented on autolysis/__init__.py:
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
+    parameters = ('p', 'chi2', 'dof', 'V')
+    for index, column in itertools.combinations(columns, 2):
+        agg_col = values if values in data.columns else column
+        agg_func = bz.count(data[agg_col]) if agg_col == column else bz.sum(data[agg_col])
```
line too long, could we multi-line it?

pratapvardhan replied:
It's 90 char long, how does it pass, what are you cooking? =)

skathirmani replied:
@pratapvardhan : Anand has changed the line char limit to 99.. Checkout [here](https://github.com/gramener/autolysis/blob/crosstabs/setup.cfg#L2)

pratapvardhan replied:
Duh, please ignore all other line-length comments, Apologies. Look into the docstring thing.

skathirmani commented:
@pratapvardhan: It passes flake8

pratapvardhan commented on autolysis/__init__.py:
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
+        result[(index, column)] = {'observed': r['observed'],
+                                   'expected': r['expected'],
+                                   'stats': {param: r[param] for param in parameters}
+                                   }
```
Can be in-line

```python
        result[(index, column)] = {
            'observed': r['observed'],
            'expected': r['expected'],
            'stats': {param: r[param] for param in parameters}
            }
```

pratapvardhan commented on autolysis/__init__.py:
```diff
@@ -211,3 +212,149 @@ def groupmeans(data, groups, numbers,
+        result = autolyse.crosstab(
+            index=data['city'],
+            column=data['language'],
+            values=data['people'])
```
should be `autolyse._crosstab`, but would this function be public?

pratapvardhan commented:
Thanks @sanand0 and  @skathirmani . Merging into dev.
